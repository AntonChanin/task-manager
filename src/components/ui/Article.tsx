import { FC } from 'react';

import { BaseComponentProps } from '../../types/common';

type Props = Partial<BaseComponentProps>

const Article: FC<Props> = ({ value = '', styles }) => {
  const { className } = styles ?? { className : '' };
  return <article className={`font-sans ${className}`.trimEnd()}>{value}</article>;
};

export default Article;
