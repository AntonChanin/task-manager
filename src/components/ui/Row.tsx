import React, { FC, PropsWithChildren } from 'react';

import createClass from '../../utils/createClass';

const Row: FC<PropsWithChildren> = ({ children }) => {
    return <div className={createClass(['flex', 'justify-between'])}>{children}</div>
}

export default Row;
