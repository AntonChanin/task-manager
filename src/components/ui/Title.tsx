import { FC } from 'react'

type Props = {
  value?: string;
  className?: string;
};

const Title: FC<Props> = ({ value = '', className = '' }) => (
  <div className={`font-sans font-bold ${className}`.trimEnd()}>{value}</div>
);

export default Title;
