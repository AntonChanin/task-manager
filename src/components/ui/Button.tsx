import { FC, MouseEventHandler, PropsWithChildren } from 'react';
import tailwindcssStyles from '../../assets/styles';

import { TButton } from '../../types/ui';

const Button: FC<PropsWithChildren<Partial<TButton>>> = (props) => {
  const {
    value = '',
    className = '',
    variant = 'primary',
    theme = 'light',
    callback,
    onClick: handleClick,
    children,
  } = props;
  const defaultHandleClick: MouseEventHandler<HTMLButtonElement> = (event) => {
    event?.preventDefault();
    callback?.();
  };
  return (
    <button
      onClick={handleClick ?? defaultHandleClick}
      className={
        `${tailwindcssStyles['ui']['button'][variant](theme)} ${className}`
          .trimEnd()
      }
    >
      {value}{children}
    </button>
  );
}

export default Button;
