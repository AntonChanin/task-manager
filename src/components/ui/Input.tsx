import React, { ChangeEventHandler, FC, useState } from 'react';

import createClass from '../../utils/createClass';
import tailwindcssStyles from '../../assets/styles';
import { EditProps } from '../../types/ui';

const Input: FC<Partial<EditProps>> = (props) => {
    const {
        className,
        variant = 'primary',
        theme = 'light',
        value,
        placeholder,
        callback,
    } = props;
    const [currentValue, setNewValue] = useState(value);

    const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
        event.preventDefault();
        setNewValue(event.target.value)
        callback?.({ value: event.target.value });
    };

    return (
        <input
            onChange={handleChange}
            value={currentValue}
            placeholder={placeholder}
            className={
                createClass([
                    'w-11/12',
                    tailwindcssStyles['ui']['edit'][variant](theme),
                    `${className}`
                ])
            }
        />
    );
};

export default Input;
