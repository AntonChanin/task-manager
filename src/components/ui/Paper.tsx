import { FC, PropsWithChildren } from 'react';
import tailwindcssStyles from '../../assets/styles';

import { TPaper } from '../../types/ui';

const Paper: FC<PropsWithChildren<Partial<TPaper>>> = (props) => {
  const {
    className = 'drop-shadow-md bg-white',
    variant = 'primary',
    theme = 'light',
    children,
  } = props
  return (
    <div className={`${className} ${tailwindcssStyles['ui']['paper'][variant](theme)}`}>
      {children}
    </div>
  );
};

export default Paper;
