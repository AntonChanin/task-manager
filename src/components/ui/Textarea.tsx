import React, { ChangeEventHandler, FC, useState } from 'react';

import createClass from '../../utils/createClass';
import { EditProps } from '../../types/ui';

const Textarea: FC<Partial<EditProps>> = (props) => {
  const { className, value, placeholder, callback } = props;
  const [currentValue, setNewValue] = useState(value);

  const handleChange: ChangeEventHandler<HTMLTextAreaElement> = (event) => {
    event.preventDefault();
    setNewValue(event.target.value)
    callback?.({ value: event.target.value });
  };

  return (
    <textarea
      onChange={handleChange}
      value={currentValue}
      placeholder={placeholder}
      className={
        createClass([
          'w-11/12',
          'outline-none',
          'border',
          'm-auto',
          `${className}`
        ])
      }
    />
  );
};

export default Textarea;
