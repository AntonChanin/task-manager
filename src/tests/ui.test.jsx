import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {describe, expect, it } from '@jest/globals';


import Article from '../components/ui/Article';
import Button from '../components/ui/Button';
import Input from '../components/ui/Input';
import Paper from '../components/ui/Paper';
import Row from '../components/ui/Row';
import Textarea from '../components/ui/Textarea';
import Title from '../components/ui/Title';

const handleClick = jest.fn();

describe('UI snapshots:', () => {
    it('Article renders correctly with props', () => {
        const tree = renderer
            .create(<Article className="article" value="description"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Article renders correctly no props', () => {
        const tree = renderer
            .create(<Article />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Button renders correctly', () => {
        const tree = renderer
            .create(
                <Button className="button" value="description" onClick={handleClick} />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Button renders correctly secondary', () => {
        const tree = renderer
            .create(
                <Button variant="secondary" className="button" value="description" onClick={handleClick} />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Input renders correctly', () => {
        const tree = renderer
            .create(
                <Input
                    value={'name'}
                    variant="thirdy"
                    className="input"
                    placeholder="name input"
                    callback={handleClick}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Paper renders correctly secondary', () => {
        const tree = renderer
            .create(
                <Paper
                    variant="secondary"
                    className="paper"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Paper renders correctly thirdy', () => {
        const tree = renderer
            .create(
                <Paper
                    variant="thirdy"
                    className="paper"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Row renders correctly', () => {
        const tree = renderer
            .create(
                <Row
                    variant="secondary"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Textarea renders correctly primary', () => {
        const tree = renderer
            .create(
                <Textarea
                    value="description"
                    variant="primary"
                    placeholder=""
                    callback={handleClick}
                    className="textarea"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Textarea renders correctly secondary', () => {
        const tree = renderer
            .create(
                <Textarea
                    value="description"
                    variant="secondary"
                    placeholder=""
                    callback={handleClick}
                    className="textarea"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Textarea renders correctly thirdy', () => {
        const tree = renderer
            .create(
                <Textarea
                    value="description"
                    variant="thirdy"
                    placeholder=""
                    callback={handleClick}
                    className="textarea"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Title renders correctly with props', () => {
        const tree = renderer
            .create(
                <Title value="title" className="title" />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Title renders correctly no props', () => {
        const tree = renderer
            .create(
                <Title />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});