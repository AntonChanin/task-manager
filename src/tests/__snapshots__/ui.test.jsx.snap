// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI snapshots: Article renders correctly no props 1`] = `
<article
  className="font-sans"
/>
`;

exports[`UI snapshots: Article renders correctly with props 1`] = `
<article
  className="font-sans article"
>
  description
</article>
`;

exports[`UI snapshots: Button renders correctly 1`] = `
<button
  className="bg-gray-400 text-white border-none outline-none m-2 hover:bg-gray-600 button"
  onClick={[MockFunction]}
>
  description
</button>
`;

exports[`UI snapshots: Button renders correctly secondary 1`] = `
<button
  className="text-gray-400 border-gray-400 border outline-none m-2 hover:text-gray-600 border-gray-600 button"
  onClick={[Function]}
>
  description
</button>
`;

exports[`UI snapshots: Input renders correctly 1`] = `
<input
  className="w-11/12 w-full outline-none border-none m-auto input"
  onChange={[Function]}
  placeholder="name input"
  value="name"
/>
`;

exports[`UI snapshots: Paper renders correctly secondary 1`] = `
<div
  className="paper border-none p-2 mb-4 text-start"
/>
`;

exports[`UI snapshots: Paper renders correctly thirdy 1`] = `
<div
  className="paper border border-black p-2 mb-4 text-start"
/>
`;

exports[`UI snapshots: Row renders correctly 1`] = `
<div
  className="flex justify-between"
/>
`;

exports[`UI snapshots: Textarea renders correctly primary 1`] = `
<textarea
  className="w-full outline-none border m-auto textarea"
  onChange={[Function]}
  placeholder=""
  value="description"
/>
`;

exports[`UI snapshots: Textarea renders correctly secondary 1`] = `
<textarea
  className="w-full outline-none border-none border-b m-auto textarea"
  onChange={[Function]}
  placeholder=""
  value="description"
/>
`;

exports[`UI snapshots: Textarea renders correctly thirdy 1`] = `
<textarea
  className="w-full outline-none border-none m-auto textarea"
  onChange={[Function]}
  placeholder=""
  value="description"
/>
`;

exports[`UI snapshots: Title renders correctly no props 1`] = `
<div
  className="font-sans font-bold"
/>
`;

exports[`UI snapshots: Title renders correctly with props 1`] = `
<div
  className="font-sans font-bold title"
>
  title
</div>
`;
